src/Core/Ast/Parser/NikicPhpParser/FileReferenceVisitor.php

```diff
@@ -40,16 +40,28 @@ class FileReferenceVisitor extends NodeVisitorAbstract
 
     private ReferenceBuilder $currentReference;
 
+    private readonly ?string $internalTag;
+
+    /**
+     * @param array{internal_tag: ?string} $config
+     */
     public function __construct(
         private readonly FileReferenceBuilder $fileReferenceBuilder,
         private readonly TypeResolver $typeResolver,
+        array $config,
         ReferenceExtractorInterface ...$dependencyResolvers
     ) {
         $this->currentTypeScope = new TypeScope('');
         $this->lexer = new Lexer();
         $this->docParser = new PhpDocParser(new TypeParser(), new ConstExprParser());
         $this->dependencyResolvers = $dependencyResolvers;
         $this->currentReference = $fileReferenceBuilder;
+
+        if ($config['internal_tag'] ?? null) {
```

Comment:
This is no longer necessary here based on the param specification. It says the array key always exists and is of type string or null.